che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/util/OpenShiftValidator.java

```diff
@@ -0,0 +1,105 @@
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.ext.openshift.client.util;
+
+import com.google.common.base.Strings;
+
+/**
+ * Utility class, that contains validation methods for various
+ * OpenShift entities
+ *
+ * @author Michail Kuznyetsov
+ */
+public class OpenShiftValidator {
+    private static final String LABEL_NAME   = "(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?";
+    private static final String ENV_NAME     = "[A-Z]([_A-Z0-9]*[A-Z0-9])?";
+    private static final String LABEL_PREFIX = "[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*";
+    private static final String PROJECT_NAME = "[a-z]([-a-z0-9]*[a-z0-9])?";
+
+    private OpenShiftValidator() {
+    }
+
+    /**
+     * Checks whether name matches OpenShift project naming rules.
+     *
+     * @param name
+     *         OpenShift project name
+     * @return true if project name is valid, false otherwise
+     */
+    public static boolean isProjectNameValid(String name) {
+        return !(Strings.isNullOrEmpty(name)
+                 || name.length() > 63
+                 || name.length() < 2
+                 || !name.matches(PROJECT_NAME));
+    }
+
+    /**
+     *
+     * @param name
+     *         OpenShift application name
+     * @return true if application name is valid, false otherwise
+     */
+    public static boolean isApplicationNameValid(String name) {
+        return !(Strings.isNullOrEmpty(name)
+                 || name.length() > 63
+                 || !name.matches(PROJECT_NAME));
+    }
+
+    /**
+     *
+     * @param variable
+     *         OpenShift application variable name
+     * @return true if variable name is valid, false otherwise
+     */
+    public static boolean isEnvironmentVariableNameValid(String variable) {
+        //TODO redo validation if needed
+        return !(Strings.isNullOrEmpty(variable)
+                 || variable.length() > 63
+                 || !variable.matches(ENV_NAME));
+    }
+
+    /**
+     *
+     * @param variable
+     *         OpenShift application variable name
+     * @return true if variable name is valid, false otherwise
+     */
+    public static boolean isEnvironmentVariableValueValid(String variable) {
+        //TODO define validation if needed
+        return true;
+    }
+
+    /**
+     *
+     * @param label
+     *         OpenShift application label name
+     * @return true if label name is valid, false otherwise
+     */
+    public static boolean isLabelValid(String label) {
+        //TODO use one method to validate name & value
```

Comment:
validation for labels is same for it's name and value, hence i will combine it in one method.
validation for env variables is applied only to its name, hence I'll remove value validation method.